pkg/adapter/kubernetes/dogucr/DoguInstallationRepo.go

```diff
@@ -0,0 +1,133 @@
+				Message:      fmt.Sprintf("cannot load dogu CR %q as it does not exist", doguName),
+			}
+		}
+		return nil, &domainservice.InternalError{
+			WrappedError: err,
+			Message:      fmt.Sprintf("error while loading dogu CR %q", doguName),
+		}
+	}
+
+	return parseDoguCR(cr)
+}
+
+func parseDoguCR(cr *v1.Dogu) (*ecosystem.DoguInstallation, error) {
+	if cr == nil {
+		return nil, &domainservice.InternalError{
+			WrappedError: nil,
+			Message:      "Cannot parse dogu CR as it is nil",
+		}
+	}
+	// parse dogu fields
+	version, versionErr := core.ParseVersion(cr.Spec.Version)
+	namespace, _, nameErr := serializer.SplitDoguName(cr.Spec.Name)
+	err := errors.Join(versionErr, nameErr)
+	if err != nil {
+		return nil, &domainservice.InternalError{
+			WrappedError: err,
+			Message:      "Cannot load dogu CR as it cannot be parsed correctly",
+		}
+	}
+	// parse persistence context
+	persistenceContext := make(map[string]interface{}, 1)
+	persistenceContext[doguInstallationRepoContextKey] = doguInstallationRepoContext{
+		resourceVersion: cr.GetResourceVersion(),
+	}
+	return &ecosystem.DoguInstallation{
+		Namespace:          namespace,
+		Name:               cr.Name,
+		Version:            version,
+		Status:             cr.Status.Status,
+		Health:             ecosystem.HealthStatus(cr.Status.Health),
+		UpgradeConfig:      ecosystem.UpgradeConfig{AllowNamespaceSwitch: cr.Spec.UpgradeConfig.AllowNamespaceSwitch},
+		PersistenceContext: persistenceContext,
+	}, nil
+}
+
+// getPersistenceContext reads the repo-specific resourceVersion from the domain.BlueprintSpec or returns an error.
+func getPersistenceContext(ctx context.Context, spec ecosystem.DoguInstallation) (doguInstallationRepoContext, error) {
+	logger := log.FromContext(ctx).WithName("doguInstallationRepo.Update")
+	rawField, versionExists := spec.PersistenceContext[doguInstallationRepoContextKey]
+	if versionExists {
+		persistenceCtx, isContext := rawField.(doguInstallationRepoContext)
+		if isContext {
+			return persistenceCtx, nil
+		} else {
+			err := fmt.Errorf("persistence context in doguInstallation is not a 'doguInstallationRepoContext' but '%T'", rawField)
+			logger.Error(err, "does this value come from a different repository?")
+			return doguInstallationRepoContext{}, err
+		}
+	} else {
+		err := errors.New("no doguInstallationRepoContext was provided over the persistenceContext in the given doguInstallation")
+		logger.Error(err, "This is normally written while loading the doguInstallation over this repository. "+
+			"Did you try to persist a new doguInstallation with repo.Update()?")
+		return doguInstallationRepoContext{}, err
+	}
+}
+
+func (repo *doguInstallationRepo) GetAll(ctx context.Context) (map[string]*ecosystem.DoguInstallation, error) {
+	crList, err := repo.doguClient.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, &domainservice.InternalError{
+			WrappedError: err,
+			Message:      fmt.Sprintf("error while listing dogu CRs"),
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1039: unnecessary use of fmt.Sprintf